index.js

```diff
@@ -1,74 +1,121 @@
+                indent,
+                force,
                 excludes,
             } = argv;
 
             if (excludes?.length > 0) {
-                files = files.filter( file => !excludes.includes(/[^/]*$/.exec(file)[0]));
+                files = files.filter(file => !excludes.includes(/[^/]*$/.exec(file)[0]));
             }
 
-            files.forEach( async (file) => {
-                let jsonFile = await fs.readFileSync(file);
-                let jsonContent = JSON.parse(jsonFile);
-                let sortedJsonContent = sortJson(jsonContent, {});
+            let unsortedFiles = [];
+            let index = 0;
+            await Promise.all(files.map(file => {
+                index += 1;
+                const jsonFile = fs.readFileSync(file);
+                const jsonContent = JSON.parse(jsonFile);
+                const sortedJsonContent = sortJson(jsonContent, {});
+                const stringContent = JSON.stringify(jsonContent);
+                const sortedStringContent = JSON.stringify(sortedJsonContent);
+
 
                 // No CI check mode. Just sort and update the given json file locally.
+                const fileName = `${file}`.bold.dim;
+                let formattedContent = JSON.stringify(sortedJsonContent, null, indent);
                 if (dryRun === false) {
-                    exec(`sort-json ${file}`, (error) => {
-                        if (error) {
-                            console.log(`error: ${error.message}`);
-                        }
-                    });
-
-                    console.info(`${file} is sorted alphabetically!`);
-                    return 0;
+                    if (force && stringContent === sortedStringContent) {
+                        fs.writeFileSync(file, formattedContent)
+                        console.info(`[${index}] ${fileName} sorted (forced)`.blue);
+                        unsortedFiles.push(file);
+                    } else if (stringContent !== sortedStringContent) {
+                        console.info(`[${index}] ${fileName} sorted`.yellow);
+                        fs.writeFileSync(file, formattedContent)
+                        unsortedFiles.push(file);
+                    } else if (!unsortedOnly) {
+                        console.info(`[${index}] ${fileName} is already sorted`.green);
+                    }
+                    return;
                 }
 
                 // CI check mode. Compare sorted file and local file.
-                if (JSON.stringify(jsonContent) !== JSON.stringify(sortedJsonContent)) {
-                    throw new Error(`${file} not sorted alphabetically`);
+                if (stringContent !== sortedStringContent) {
+                    console.error(`[${index}] ${fileName} not sorted`.red);
+                    unsortedFiles.push(file);
+                } else if (!unsortedOnly) {
+                    console.info(`[${index}] ${fileName} is sorted`.green);
                 }
```

Comment:
It is not clear if the output is from the --dry-run so I would suggest to change the wording and maybe the colors.